src/tcgwars/logic/impl/gen4/CallOfLegends.groovy

```diff
@@ -1593,110 +788,80 @@ public enum CallOfLegends implements LogicCardInfo {
+            pcs.cards.filterByTYpe(SPECIAL_ENERGY).select("Put which special energy in the Lost Zone?").first().moveTo(opp.lostZone)
           }
           playRequirement{
+            assert opp.all.find{it.cards.filterByType(SPECIAL_ENERGY)}
           }
         };
       case LOST_WORLD_81:
-        return basicTrainer (this) {
+        return stadium (this) {
           text "This card stays in play when you play it. Discard this card if another Stadium card comes into play. If another card with the same name is in play, you can’t play this card.\nOnce during each player’s turn, if this player’s opponent has 6 or more Pokémon in the Lost Zone, the player may choose to win the game."
+          def lastTurn=0
+          def actions=[]
           onPlay {
+            actions=action("Stadium: Lost World") {
+              assert lastTurn != bg().turnCount : "You've already used Lost World this turn."
+              bc "Used Lost World."
+              lastTurn = bg().turnCount
+              bg.getGame().endGame(self.owner, WinCondition.OTHER);
+            }
           }
-          playRequirement{
+          onRemoveFromPlay{
+            actions.each { bg().gm().unregisterAction(it) }
           }
         };
       case PROFESSOR_ELM_S_TRAINING_METHOD_82:
-        return basicTrainer (this) {
-          text "You can play only one Supporter card each turn. When you play this card, put it next to your Active Pokémon. When your turn ends, discard this card.\nSearch your deck for an Evolution card, show it to your opponent, and put it into your hand. Shuffle your deck afterward."
-          onPlay {
-          }
-          playRequirement{
-          }
-        };
+        return copy(HeartgoldSoulsilver.PROFESSOR_ELM_S_TRAINING_METHOD_100, this);
       case PROFESSOR_OAK_S_NEW_THEORY_83:
-        return basicTrainer (this) {
-          text "You can play only one Supporter card each turn. When you play this card, put it next to your Active Pokémon. When your turn ends, discard this card.\nShuffle your hand into your deck. Then, draw 6 cards."
-          onPlay {
-          }
-          playRequirement{
-          }
-        };
+        return copy(HeartgoldSoulsilver.PROFESSOR_OAK_S_NEW_THEORY_101, this);
       case RESEARCH_RECORD_84:
         return basicTrainer (this) {
           text "Look at the top 4 cards of your deck and put as many of them as you like back on top of your deck in any order. Then, put the remaining cards on the bottom of your deck in any order."
           onPlay {
+            def tar = my.deck.subList(0,5)
+            def sel = tar.select(min:0, max:tar.size(), "Choose any number of cards to put on top of your deck(unselected cards will be put on the bottom of your deck)")
+            if(sel){
+              tar.remove(sel)
+              deck.setSublist(0,rearrange(sel,"Arrange the top of your deck"))
```

Comment:
```suggestion
              def rearranged = rearrange(sel, "Arrange the top of your deck")
              deck.setSublist(0, rearranged)
```
It's usually easier to read if you store the result of a method call in a variable and then use it in another method call rather than try to include it all in the same line.